assets/js/blocks/product-category/block.js

```diff
@@ -220,14 +273,14 @@ class ProductByCategoryBlock extends Component {
 							{
 								icon: 'edit',
 								title: __( 'Edit' ),
-								onClick: () => setAttributes( { editMode: ! editMode } ),
-								isActive: editMode,
+								onClick: () => this.startEditing(),
```

Comment:
This should toggle the edit mode, not just "start editing" – probably when in edit mode, this should do the same behavior as cancelling out.